repos/spack_repo/builtin/packages/lammps/package.py

```diff
@@ -415,6 +415,7 @@ class Lammps(CMakePackage, CudaPackage, ROCmPackage, PythonExtension):
         deprecated=True,
     )
 
+    depends_on("c", type="build", when="@20250612:")
```

Comment:
Did we consider simplifying this whole thing and:
```python
depends_on("c", type="build")
```
plus delete the conditional dependency below on the C language? This will simplify the maintenance of the package, without imposing more constraints, practically speaking (usually, if there's a C++ compiler available, there's also the C compiler of the same toolchain)